### PR TITLE
Delete default route on vpc creation

### LINKED ISSUE #1401: Always delete default route on VPC module and create an explicit one
To make it easier to add/remove a default route (to the default gateway) on the VPC module, we could always delete the auto-created default route and create a TF managed one that can be created/deleted via flag.

### PR DESCRIPTION
**Pull request by juliocc**
Fixes #1401

### CODE REVIEW

**Comment by juliocc** [comment]
I myself am ambivalent. This will break the VPCs of anyone upgrading from previous versions.

File: modules/net-vpc/routes.tf [hunk 1]
```diff
@@ -17,6 +17,15 @@
 # tfdoc:file:description Route resources.
 
 locals {
+  _default_route = var.vpc_create && var.delete_default_routes_on_create ? {} : {
```
Comment (wiktorn) [inline modules/net-vpc/routes.tf:20]: This will create default_route also when `var.vpc_create` is set to false. Is this intended?

Comment (juliocc) [reply modules/net-vpc/routes.tf]: should be fixed now. PTAL

**Comment by wiktorn** [comment]
Tested migration scenario. So the end result is that I'm left with default route created by default and additional default route created by module.

It looks like the `delete_default_routes_on_create` is implemented in provider and only when creating the network. So the proper migration path is to remove manually the auto-created one.

I think it will only affect those, who already removed the default route manually.

**Comment by ludoo** [comment]
TBH I don't see the need to this: it covers a use case which is practically non-existent (forget to set delete default route at vpc creation), and can be addressed with a single click in the console.

The fact that it creates an apocalypse for all existing users of the module is a huge red flag, and the two combines make me *not* want this change at all.

**Comment by ludoo** [comment]
discussed it with Julio, closing this

**Pull request closed without merging**